### PR TITLE
fix: tx submisson

### PR DESCRIPTION
**Pull request by rflechtner**
## fixes KILTProtocol/ticket#1167
Switching to `submitTxWithResign` fixes the issue of extrinsic errors going unnoticed (this uses defaults for `rejectOn`).

## How to test:
Try different transactions that must fail - attesting on a revoked delegationNode etc.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by tjwelde** [comment]
This is nice and all, but shouldn't we also fix the root issue in the sdk?

**Comment by rflechtner** [comment]
> This is nice and all, but shouldn't we also fix the root issue in the sdk?

Sure, but that is what I understood https://github.com/KILTprotocol/ticket/issues/1184 to be about?
This change here was meant to quickly get the demo client up and running again - it was not meant to linger in review for weeks. We have several bugs in the backlog that I suspect many of them to be related to this (https://github.com/KILTprotocol/ticket/issues/1160, https://github.com/KILTprotocol/ticket/issues/1215).

File: src/components/DelegationNode/DelegationNode.tsx [hunk 1]
```diff
@@ -276,6 +276,13 @@ class DelegationNode extends React.Component<Props, State> {
 
     const hashes = await delegation.getAttestationHashes()
 
+    if (hashes.length < 1) {
```
Comment (LeonFLK) [inline src/components/DelegationNode/DelegationNode.tsx:279]: Lets maybe not include these changes in a temp fix PR, after we talk tomorrow, these changes might need to be reverted again.

File: src/services/DelegationsService.ts [hunk 2]
```diff
@@ -23,40 +23,37 @@ class DelegationsService {
...
     const selectedIdentity = Wallet.getSelectedIdentity(
       persistentStoreInstance.store.getState()
     )?.identity
 
     if (!selectedIdentity) {
       throw new Error('No selected Identity')
     }
-    return delegation.store(selectedIdentity, signature)
+    const tx = await delegation.store(selectedIdentity, signature)
+    return BlockchainUtils.submitTxWithReSign(tx, selectedIdentity, {
```
Comment (LeonFLK) [inline src/services/DelegationsService.ts:54]: drying code

File: src/components/DelegationNode/DelegationNode.tsx [hunk 3]
```diff
@@ -276,6 +276,13 @@ class DelegationNode extends React.Component<Props, State> {
...
+      notify(
+        <span>No attestations associated with this Delegation</span>,
+        false
+      )
+    }
+
```
Comment (LeonFLK) [inline src/components/DelegationNode/DelegationNode.tsx:285]:
```suggestion
```

**Review by LeonFLK — COMMENTED** [review]
small nitpick, otherwise LGTM

**Comment by tjwelde** [comment]
@rflechtner yes, i see your argument. please go ahead

File: src/components/DelegationNode/DelegationNode.tsx (hunk 1 above)
Comment (rflechtner) [reply src/components/DelegationNode/DelegationNode.tsx:279]: That's true, but I doubt we'd do a revert of the PR. More likely we'd change all occurrences of `BlockchainUtils.submitTxWithReSign` to whatever we come up in the redesign.
I'd say if restructured some of the code to be a little cleaner here, why would we want to throw that away later.

Comment (rflechtner) [reply src/components/DelegationNode/DelegationNode.tsx:279]: but I can of course pull it in a separate PR, doesn't hurt

**Review by LeonFLK — APPROVED** [review]
LGTM

File: src/components/DelegationNode/DelegationNode.tsx (hunk 3 above)
Comment (rflechtner) [reply src/components/DelegationNode/DelegationNode.tsx:285]: pulled this out to https://github.com/KILTprotocol/demo-client/pull/345, will remove here now

**Pull request merged**